universal-login-react/src/ui/UFlow/DashboardModal.tsx

```diff
@@ -0,0 +1,150 @@
+import {TransferService, WalletService} from '@universal-login/sdk';
+import React, {useState} from 'react';
+import {asTransferDetails, GasParameters, TransferDetails} from '@universal-login/commons';
+import {Route, Switch} from 'react-router';
+import {TopUp} from '../TopUp/TopUp';
+import {TransferAmount} from '../Transfer/Amount/TransferAmount';
+import {TransferRecipient} from '../Transfer/Recipient/TransferRecipient';
+import {Devices} from './Devices/Devices';
+import BackupCodes from '../BackupCodes/BackupCodes';
+import {cast} from '@restless/sanitizers';
+import {InvalidTransferDetails} from '../../core/utils/errors';
+import {WaitingForTransaction} from '../commons/WaitingForTransaction';
+import {DialogWrapper} from './DialogWrappers/DialogWrapper';
+import {SubDialogWrapper} from './DialogWrappers/SubDialogWrapper';
+import {ModalWrapper} from '../Modals/ModalWrapper';
+import {Funds} from './Funds';
+import {NewDeviceMessage} from './Devices/NewDeviceMessage';
+
+export interface DashboardModalProps {
+  walletService: WalletService;
+  onClose: () => void;
+}
+
+export const DashboardModal = ({walletService, onClose}: DashboardModalProps) => {
+  const {relayerConfig} = walletService.sdk;
+
+  const deployedWallet = walletService.getDeployedWallet();
+  const {contractAddress, sdk, name} = deployedWallet;
+
+  const [transferDetails, setTransferDetails] = useState<Partial<TransferDetails>>({transferToken: sdk.tokensDetailsStore.tokensDetails[0].address} as TransferDetails);
+  const selectedToken = sdk.tokensDetailsStore.getTokenByAddress(transferDetails.transferToken!);
+  const [transactionHash, setTransactionHash] = useState('');
+
+  const notice = sdk.getNotice();
```

Comment:
This shouldn't be in useEffect?

Reply:
No, `getNotice` just reads a variable so no need for `useEffect` here